src/foam/nanos/bench/BenchmarkRunner.java

```diff
@@ -145,43 +152,65 @@ public String getResult() {
 
   @Override
   public void execute(final X x) {
-    Logger logger = (Logger) x.get("logger");
-    if ( logger != null ) {
-      logger.info(this.getClass().getSimpleName(), "execute", test_.getClass().getSimpleName());
+    Logger log = (Logger) x.get("logger");
+    if ( log != null ) {
+      log = new foam.nanos.logger.StdoutLogger();
     }
+    final Logger logger = new PrefixLogger(new String[] { test_.getClass().getSimpleName() }, log);
 
-    int availableThreads = Runtime.getRuntime().availableProcessors();
+    AppConfig config = (AppConfig) x.get("appConfig");
+    if ( config.getMode() == Mode.PRODUCTION ) {
+      logger.warning("Script execution disabled in PRODUCTION");
+      return;
+    }
+    logger.info("execute", test_.getClass().getSimpleName());
+
+    int availableThreads = Math.min(Runtime.getRuntime().availableProcessors(), getThreadCount());
     int run = 1;
-    if ( runPerThread_ ) {
-      threadCount_ = 1;
-      if ( reverseThreads_ ) {
-        threadCount_ = availableThreads;
-      }
+    int threads = 1;
+    if ( ! getRunPerThread() ) {
+      availableThreads = 1;
+    } else if ( reverseThreads_ ) {
+        threads = availableThreads;
```

Comment:
You can't reverseThreads unless you enable getRunPerThread()?

Reply:
when runPerThread is false - we only do one run.  
But what's missing is  runPerThread = false, but still do one run with all threads.